jme3-alloc/src/main/java/com/jme3/alloc/util/NativeBufferUtils.java

```diff
@@ -0,0 +1,116 @@
+package com.jme3.alloc.util;
+
+import java.io.IOException;
+import java.nio.Buffer;
+import com.jme3.alloc.util.loader.NativeImageLoader;
+
+/**
+ * A jvm low-level implementation to the native GNU-Standard-Library [stdlib].
+ * 
+ * @author pavl_g
+ */
+public final class NativeBufferUtils {
+
+    static {
+        try {
+            /* extracts and loads the system specific library */
+            NativeImageLoader.loadLibrary();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private NativeBufferUtils() {
+    }
+
+    /**
+     * Allocates a buffer of size bytes.
+     * 
+     * @param size the size of the newly created buffer in bytes (max = 32-bit or 4 bytes)
+     * @return a new java-nio buffer object
+     */
+    public static native Buffer memoryAlloc(final long size);
+
+    /**
+     * Copies the value [value] into each of the first [size] bytes of the object beginning at the 
+     * memory address of [buffer].
+     * 
+     * @param buffer a buffer block to start from
+     * @param value the value to set the memory blocks with starting from the buffer
+     * @param size the size to extend after the buffer address
+     * @return the buffer block
+     */
+    public static native void memorySet(final Buffer buffer, final int value, final long size); 
```

Comment:
I wonder why this method is needed, can't we set this from java side?

Reply:
I want to build a full api, i am using GNU libc and the nio jni documentation, so i am still not sure if this is actually functional in jni.

Reply:
I will provide a testcase for this method, it provides the capability to set all the buffer byte elements with a [value], it can be done through java with a for-loop, but i prefer to keep the native implementation anyway, the native implementation uses the memory address and the size to define the buffer boundaries.